mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/MailboxMergingTask.java

```diff
@@ -44,21 +53,69 @@ public String getOldMailboxId() {
         public String getNewMailboxId() {
             return newMailboxId.serialize();
         }
+
+        public long getTotalMessageCount() {
+            return totalMessageCount;
+        }
+
+        public long getMessageMovedCount() {
+            return messageMovedCount;
+        }
+
+        public long getMessageFailedCount() {
+            return messageFailedCount;
+        }
+    }
+
+    public static class Context {
+        private final long totalMessageCount;
+        private final AtomicLong totalMessageMoved;
+        private final AtomicLong totalMessageFailed;
+
+        public Context(long totalMessagesCount) {
+            this.totalMessageCount = totalMessagesCount;
+            this.totalMessageMoved = new AtomicLong(0L);
+            this.totalMessageFailed = new AtomicLong(0L);
+        }
+
+        public long getTotalMessageCount() {
+            return totalMessageCount;
+        }
+
+        public long getMessageMovedCount() {
+            return totalMessageMoved.get();
+        }
+
+        public long getMessageFailedCount() {
+            return totalMessageFailed.get();
+        }
+
+        public void incrementMovedCount() {
+            totalMessageMoved.incrementAndGet();
+        }
+
+        public void incrementFailedCount() {
+            totalMessageFailed.incrementAndGet();
+        }
     }
 
     private final MailboxMergingTaskRunner taskRunner;
     private final CassandraId oldMailboxId;
     private final CassandraId newMailboxId;
+    private final Context context;
 
-    public MailboxMergingTask(MailboxMergingTaskRunner taskRunner, CassandraId oldMailboxId, CassandraId newMailboxId) {
+    public MailboxMergingTask(MailboxMergingTaskRunner taskRunner, CassandraMailboxCounterDAO counterDAO, CassandraId oldMailboxId, CassandraId newMailboxId) {
```

Comment:
I would give the number of messages directly to the constructor instead of the DAO.